Projects/Server/Gumps/New/Components/GumpBuilder.cs

```diff
@@ -0,0 +1,698 @@
+/*************************************************************************
+ * ModernUO                                                              *
+ * Copyright 2019-2023 - ModernUO Development Team                       *
+ * Email: hi@modernuo.com                                                *
+ * File: GumpBuilder.cs                                                  *
+ *                                                                       *
+ * This program is free software: you can redistribute it and/or modify  *
+ * it under the terms of the GNU General Public License as published by  *
+ * the Free Software Foundation, either version 3 of the License, or     *
+ * (at your option) any later version.                                   *
+ *                                                                       *
+ * You should have received a copy of the GNU General Public License     *
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>. *
+ *************************************************************************/
+
+using Server.Gumps.Components.Interpolation;
+using Server.Gumps.Enums;
+using Server.Gumps.Interfaces;
+using Server.Network;
+using System;
+using System.Buffers;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.IO.Compression;
+using System.Runtime.CompilerServices;
+using System.Text;
+using static Server.Gumps.Components.Interpolation.GumpInterpolatedStringHandler;
+using static Server.Gumps.Components.GumpBuilderExtensions;
+
+namespace Server.Gumps.Components
+{
+    public ref struct GumpBuilder<T>
+        where T : struct, IStringsHandler
+    {
+        private static readonly byte[] _layoutBuffer = LayoutBuffer;
```

Comment:
We need to find out if we can use statics in generic ref structs. We may have to just use `LayoutBuffer` directly or use `STArrayPool`.

Reply:
That is simply a static "local" variable that refers to the same global buffer.
If you prefer we can reference the global variable directly, if we want to maintain the static global buffer, or use the rented buffer.
The actual logic is similar to the old gump approach with a huge global static buffer.